cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/Diffs.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.operations.cluster;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.extensions.StatefulSet;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiPredicate;
+
+import static io.strimzi.controller.cluster.resources.AbstractCluster.containerEnvVars;
+
+class Diffs {
```

Comment:
The name of this class is fairly generic. But:
-  It seems to have some Statefulset specifics. Do you think it can be used as a generic class?
- If I understand the code correctly, you patch always everything, but this Diff is actually relevant only for the rolling update of StatefulSets, right? So, is Diff the right name?

Reply:
* There are things which could also, in principle, apply to deployments, we just have no need for that currently, AFAICS

* It's probably not the right name. I also umm-ed and ahh-ed about whether this was better put in with models (`KafkaCluster` and `ZookeeperCluster`) or the Operations.